### PR TITLE
guttermarker improvements

### PR DESCRIPTION
**Pull request by djusv**
- removed: unused source file: PreferencesManager.js
- fixed: gutter order sometimes conflicting with GitGutter
- fixed: show error icon on top when the line has both warning and error
- feature: preference to hide error/warning icons in gutter

### CODE REVIEW

File: src/GutterManager.js [hunk 1]
```diff
@@ -101,7 +128,13 @@ define(function () {
 
   function GutterManager() {
 
-    function setGutterMarkers(markers) {
+    function setGutterMarkers(markers, option) {
+      if (option && option.error === false) {
```
Comment (zaggino) [inline src/GutterManager.js]: this should be `option && typeof option.error !== 'undefined'`

File: src/GutterManager.js [hunk 2]
```diff
@@ -101,7 +128,13 @@ define(function () {
...
+        showGutterError = option.error === true;
+      }
+      if (option && option.warning === false) {
```
Comment (zaggino) [inline src/GutterManager.js]: same check as commented above

**Comment by zaggino** [comment]
i've fixed my comments and published as 2.4.0

**Comment by djusv** [comment]
@zaggino Thank you

**Pull request merged**